src/compiler/checker.ts

```diff
@@ -25756,6 +25776,37 @@ namespace ts {
             return prop !== undefined && prop.valueDeclaration && hasSyntacticModifier(prop.valueDeclaration, ModifierFlags.Static);
         }
 
+        function getSuggestedLibForNonexistentProperty(missingProperty: string, containingTypeString: string) {
+            const containingType = generalizeTypeString(containingTypeString);
+            const allNewFeatures = getScriptTargetFeatures();
+            const libTargets = Object.keys(allNewFeatures);
+            for (const libTarget of libTargets) {
+                const newFeaturesOfLib = allNewFeatures[libTarget];
+                const newFeaturesOfContainingType = newFeaturesOfLib[containingType];
+                if (newFeaturesOfContainingType) {
+                    for (const newFeature of newFeaturesOfContainingType) {
+                        const isNewFeatureTheMissingProperty = newFeature === missingProperty;
+                        if (isNewFeatureTheMissingProperty) {
+                            return libTarget;
+                        }
+                    }
+                }
+            }
+
+            function generalizeTypeString(typeString: string){
+                if (typeString.includes("[]")) {

```

Comment:
You really want to see if you have a single TypeReference and get the text of the name of its declaration.

Reply:
This makes a lot more sense, thanks